mapie/quantile_regression.py

```diff
@@ -0,0 +1,414 @@
+        """
+        if estimator is None:
+            return QuantileRegressor(solver="highs")
+        if not (hasattr(estimator, "fit") and hasattr(estimator, "predict")):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a regressor with fit and predict methods."
+            )
+        name_estimator = estimator.__class__.__name__
+        if name_estimator == "QuantileRegressor":
+            return estimator
+        else:
+            if name_estimator in self.quantile_estimator_params:
+                param_estimator = estimator.get_params()
+                if (
+                    self.quantile_estimator_params[name_estimator]["loss_name"]
+                    in param_estimator
+                ):
+                    if param_estimator[
+                        self.quantile_estimator_params[
+                            name_estimator]["loss_name"]] != "quantile":
+                        raise ValueError(
+                            "You need to set the loss/metric of your base"
+                            + " model to quantile."
+                        )
+                    else:
+                        if (
+                            self.quantile_estimator_params[
+                                name_estimator
+                                ]["alpha_name"]
+                            in param_estimator
+                        ):
+                            return estimator
+                        else:
+                            raise ValueError(
+                                "The matching parameter alpha_name for"
+                                + " estimator does not exist."
+                            )
+                else:
+                    raise ValueError(
+                        "The matching parameter loss_name for"
+                        + " estimator does not exist."
+                    )
+            else:
+                raise ValueError(
+                    "The base model does not seem to be accepted"
+                    + " by MapieQuantileRegressor."
+                )
+
+    def _check_cv(
+        self,
+        cv: Optional[str] = None
+    ) -> str:
+        """
+        Performs a check on the cv.
+
+        Parameters
+        ----------
+        cv : Optional[str], optional
+           cv to check, by default ``None``.
+
+        Returns
+        -------
+        str
+            cv itself or a default "split".
+
+        Raises
+        ------
+        ValueError
+            Raises an error if the cv is anything else but the method "split".
+            Only the split method has been implemented.
+        """
+        if cv is None:
+            return "split"
+        if cv == "split":
+            return cv
+        else:
+            raise ValueError(
+                "Invalid cv method."
+            )
+
+    def fit(
+        self,
+        X_train: ArrayLike,
```

Comment:
I would modify the order to be more consistent with the standard `fit` method : `X_train`, `y_train`, `X_calib`, `y_calib`.